src/cargo/core/workspace.rs

```diff
@@ -201,6 +201,17 @@ impl<'cfg> Workspace<'cfg> {
         }.parent().unwrap()
     }
 
+    /// Returns the root path of this workspace.
+    ///
+    /// That is, this returns the path of the directory containing the
+    /// `Cargo.toml` which is the root of this workspace.
+    pub fn workspace_dir(&self) -> Option<&Path> {
+        match self.root_manifest {
```

Comment:
I thought @Shepmaster said in #3946, that it's best it's not present if it's not a  cargo workspace project?